amaranth/hdl/ast.py

```diff
@@ -377,6 +377,9 @@ def __getitem__(self, key):
         else:
             raise TypeError(f"Cannot index value with {key!r}")
 
+    def __contains__(self, other):
+        raise TypeError(f"'{self.__class__.__name__}' object is not a container")
```

Comment:
Maybe something like `Cannot use 'in' with an Amaranth value`? That would match our other messages.